File: spring-mvc-plain-java-example/src/main/java/de/cfranzen/examples/webflux/Customer.java
package de.cfranzen.examples.webflux;


import org.springframework.data.annotation.Id;

import java.util.Objects;

public class Customer {

    @Id
    private Long id;
    private final String firstName;
    private final String lastName;

    public Customer(String firstName, String lastName) {
        this.firstName = Objects.requireNonNull(firstName, "Firstname of customer might not be null");
        this.lastName = Objects.requireNonNull(lastName, "Lastname of customer might not be null");
    }

    public Long getId() {
        return id;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    @Override
    public String toString() {
        return "Customer{" + "id=" + id + ", firstName='" + firstName + '\'' + ", lastName='" + lastName + '\'' + '}';
    }
}

File: spring-webflux-kotlin-example/src/main/kotlin/de/cfranzen/examples/webflux/kotlin/Customer.kt
package de.cfranzen.examples.webflux.kotlin

import org.springframework.data.annotation.Id

data class Customer(
    val firstName: String,
    val lastName: String
) {
    @Id
    private var id: Long = 0

    fun getId() = id

    override fun toString(): String {
        return "Customer{id=$id, firstName='$firstName', lastName='$lastName'}"
    }
}